src/orc2/mod.rs

```diff
@@ -0,0 +1,2186 @@
+        self.as_ref().fmt(f)
+    }
+}
+
+#[llvm_versions(13.0..=latest)]
+impl Drop for SymbolStringPoolEntries {
+    fn drop(&mut self) {
+        unsafe {
+            LLVMOrcDisposeSymbols(self.entries);
+        }
+    }
+}
+
+#[llvm_versions(13.0..=latest)]
+#[derive(Debug)]
+pub struct LazyCallThroughManager {
+    lazy_call_through_manager: LLVMOrcLazyCallThroughManagerRef,
+}
+
+#[llvm_versions(13.0..=latest)]
+#[derive(Debug)]
+pub struct IndirectStubsManager {
+    indirect_stubs_manager: LLVMOrcIndirectStubsManagerRef,
+}
+
+#[llvm_versions(12.0..=latest)]
+#[derive(Debug)]
+#[repr(transparent)]
+pub struct LookupState {
+    lookup_state: LLVMOrcLookupStateRef,
+}
+
+#[llvm_versions(12.0..=latest)]
+impl LookupState {
+    unsafe fn new(lookup_state: LLVMOrcLookupStateRef) -> Self {
+        assert!(!lookup_state.is_null());
+        LookupState { lookup_state }
+    }
+}
+
+#[llvm_versions(12.0..=latest)]
+#[derive(Debug)]
+#[repr(transparent)]
+pub struct LookupKind {
+    lookup_kind: LLVMOrcLookupKind,
+}
+
+#[llvm_versions(12.0..=latest)]
+impl LookupKind {
+    fn new(lookup_kind: LLVMOrcLookupKind) -> Self {
+        LookupKind { lookup_kind }
+    }
+}
+
+#[llvm_versions(12.0..=latest)]
+#[llvm_enum(LLVMOrcJITDylibLookupFlags)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub enum JITDylibLookupFlags {
+    #[llvm_variant(LLVMOrcJITDylibLookupFlagsMatchExportedSymbolsOnly)]
+    MatchExportedSymbolsOnly,
+    #[llvm_variant(LLVMOrcJITDylibLookupFlagsMatchAllSymbols)]
+    MatchAllSymbols,
+}
+
+#[llvm_versions(12.0..=latest)]
+#[derive(Debug)]
+pub struct CLookupSet {
+    set: LLVMOrcCLookupSet,
+    size: usize,
+}
+
+#[llvm_versions(12.0..=latest)]
+impl CLookupSet {
+    unsafe fn from_raw_parts(set: LLVMOrcCLookupSet, size: usize) -> Self {
+        assert!(!set.is_null());
+        CLookupSet { set, size }
+    }
+}
+
+#[llvm_versions(12.0..=latest)]
+impl Deref for CLookupSet {
+    type Target = [CLookupSetElement];
+
+    fn deref(&self) -> &Self::Target {
+        unsafe { slice::from_raw_parts(transmute(self.set), self.size) }
+    }
+}
+
+#[llvm_versions(12.0..=latest)]
+impl AsRef<[CLookupSetElement]> for CLookupSet {
+    fn as_ref(&self) -> &[CLookupSetElement] {
+        self.deref()
+    }
+}
+
+#[llvm_versions(12.0..=latest)]
+impl AsMut<[CLookupSetElement]> for CLookupSet {
+    fn as_mut(&mut self) -> &mut [CLookupSetElement] {
+        unsafe { slice::from_raw_parts_mut(transmute(self.set), self.size) }
+    }
+}
+
+#[llvm_versions(12.0..=latest)]
+#[derive(Debug)]
+#[repr(transparent)]
+pub struct CLookupSetElement {
+    element: LLVMOrcCLookupSetElement,
+}
+
+#[llvm_versions(12.0..=latest)]
+impl CLookupSetElement {
+    pub fn get_name(&self) -> SymbolStringPoolEntry {
+        unsafe { SymbolStringPoolEntry::new(self.element.Name) }
+    }
+
+    pub fn get_flags(&self) -> SymbolLookupFlags {
+        self.element.LookupFlags.clone().into()
```

Comment:
This fails to compile for me, is the `llvm_enum` macro supposed to provide a `From` implementation?